packages/engine/lib/stateful/src/message/outbound.rs

```diff
@@ -215,20 +215,16 @@ impl Message {
         if let Some(serde_json::Value::String(kind /* is a &String */)) = value.get("type") {
             // since all keys stored inside the system message types are lower case, we should also
             // lowercase the kind here too
-            let kind: &String = &kind.to_ascii_lowercase();
-            // contains needs a &str, our type is a &String, so
-            // 1. *&String -> String
-            // 2. *String -> &str
-            // 3. (wrap in a temp ref, for std::borrow::Borrow), thus &**
-            if is_system_message(&**kind) {
+            let kind = kind.to_ascii_lowercase();
+            if is_system_message(&kind) {
```

Comment:
I wonder if this was on purpose that `kind` is used for the `InvalidMessageType`

Reply:
From the comment below I'd say it wasn't.

Reply:
I think, this is the correct approach: 81e32c717f3e7088ea63d572df17f40d87462eb3
Happy to revert if I'm wrong.

Reply:
I don't think it should matter but looks fine to me